### PR TITLE
support eaf3.0 and enable scheme compliance tests

### LINKED ISSUE #29: EAF 3.0 file format support
Currently only 2.7 and 2.8 are officially _supported_ and a warning is given when another version is imported.

The scheme is available here: http://www.mpi.nl/tools/elan/EAFv3.0.xsd

A human readable explanation is available here:
https://www.mpi.nl/tools/elan/EAF_Annotation_Format_3.0_and_ELAN.pdf

It should be added to the _supported_ list after checking and implementing the differences in spec.

### PR DESCRIPTION
**Pull request by dopefishh**
Closes #29

The sample 3.0 file does not yet use one of the extra features, but they aren't really supported anyway.

### CODE REVIEW

**Comment by dopefishh** [comment]
@xrotwang can you have a look at the tests that use files? In the previous setup the tests were supposed to be executed from the root of the repo but I think this is not the case anymore.

**Comment by xrotwang** [comment]
will do

File: .gitignore [hunk 1]
```diff
@@ -10,9 +10,7 @@ lib64
 pyvenv.cfg
 
 # Test files or temporary data files
-*.TextGrid
 *.pfsx
-*.eaf
```
Comment (xrotwang) [inline .gitignore:15]: When using `pytest`'s `tmp_path` fixture, we can make sure that temporary files are not written to the repository.

File: test/conftest.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import pathlib
+
+import pytest
+
+
+@pytest.fixture
+def test_dir():
+    return pathlib.Path(__file__).parent
```
Comment (xrotwang) [inline test/conftest.py:8]: All fixtures defined in `conftest.py` are available in all tests.

File: test/test_elan.py [hunk 3]
```diff
@@ -1143,42 +1139,26 @@ def test_add_nested_reference_annotations(self):
...
 
-if __name__ == '__main__':
-    unittest.main()
+
+@pytest.mark.parametrize(
+    'eaf,schema',
+    [
+        ('sample_2.8.eaf', 'EAFv2.8.xsd'),
+        ('sample_2.7.eaf', 'EAFv2.8.xsd'),
+        #('sample_3.0.eaf', 'EAFv3.0.xsd'),  # FIXME: sample file not there yet
```
Comment (xrotwang) [inline test/test_elan.py:1154]: It seems `sample_3.0.eaf` isn't pushed yet.

Comment (dopefishh) [reply test/test_elan.py:1154]: I've added this

**Pull request merged**